compiler/src/main/java/com/juul/pommel/compiler/FunctionSoloModuleGenerator.kt

```diff
@@ -18,8 +19,43 @@ internal class FunctionSoloModuleGenerator : SoloModuleGenerator {
         return element.kind == ElementKind.METHOD
     }
 
+    // object classes and companion objects can be deeply nested
+    // this is way to get the full chained function name
+    // e.g. Sample.Module.Pommel.baseUrl()
```

Comment:
Nit: should this be a doc comment? `/** */`

Reply:
@cedrickcooke added a doc comment with example for a better explanation as to what this function does